restapi/clusters.go

```diff
@@ -0,0 +1,259 @@
+// This file is part of MinIO Console Server
+// Copyright (c) 2020 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package restapi
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/minio/m3/cluster"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/swag"
+	"github.com/minio/m3/models"
+	"github.com/minio/m3/restapi/operations"
+	"github.com/minio/m3/restapi/operations/admin_api"
+	operator "github.com/minio/minio-operator/pkg/apis/operator.min.io/v1"
+	operatorClientset "github.com/minio/minio-operator/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func registerClusterHandlers(api *operations.M3API) {
+	// Add Cluster
+	api.AdminAPICreateClusterHandler = admin_api.CreateClusterHandlerFunc(func(params admin_api.CreateClusterParams) middleware.Responder {
+		err := getClusterCreatedResponse(params)
+		if err != nil {
+			return admin_api.NewCreateClusterDefault(500).WithPayload(&models.Error{Code: 500, Message: swag.String(err.Error())})
+		}
+		return admin_api.NewCreateClusterCreated()
+	})
+}
+
+func getClusterCreatedResponse(params admin_api.CreateClusterParams) error {
+
+	if params.Body.Name == nil || *params.Body.Name == "" {
+		return errors.New("Name cannot be empty")
+	}
+
+	var re = regexp.MustCompile(`^[a-z0-9-]{3,63}$`)
+	if !re.MatchString(*params.Body.Name) {
+		return errors.New("Invalid name. Must not contain spaces or non alpha-numerical characters")
+	}
+
+	minioImage := params.Body.Image
+	if minioImage == "" {
+		minImg, err := getLatestMinIOImage()
+		if err == ErrCantDetermineMinIOImage {
+			return errors.New("can't determine the MinIO Version. please specify one")
+		} else if err != nil {
+			return err
+		}
+		minioImage = *minImg
+	}
+
+	if params.Body.VolumeConfiguration == nil {
```

Comment:
if this is required, set it on the yaml

Reply:
it's not required, objects if not set on the payload are received as null, this parameter is optional.

Reply:
why are you returning an error if it is not received? that makes it required?

Reply:
hahaha you are correct, I'm removing all this checks and only keeping them for the optional parameters